go-client/client/fieldmanager.go

```diff
@@ -0,0 +1,368 @@
+		}
+	}()
+
+	for {
+		select {
+		case <-fme.chanClose:
+			return
+		case req := <-fme.chanOpenTables:
+			req.out <- fme.listOpenableTables()
+		case req := <-fme.chanGetTable:
+			tbl, ok := fme.tables[req.id]
+			req.out <- respGetTable{table: tbl, ok: ok}
+		case req := <-fme.chanFMExec:
+			changes, err := req.f()
+			msg := respFMExec{changes: changes, err: err}
+			// The ExecAndUpdate call will handle the error for us.
+			req.out <- msg
+			if err == nil {
+				fme.handleFieldChanges(changes)
+			}
+		case req := <-fme.chanFetchRepeating:
+			if fme.fs != nil {
+				fme.fs.Close()
+				fme.fs = nil
+			}
+
+			if fme.chanFetchRepeatingError != nil {
+				close(fme.chanFetchRepeatingError)
+				fme.chanFetchRepeatingError = nil
+			}
+
+			fme.fs = req.fs
+			fme.chanFetchRepeatingError = req.chanError
+			// TODO: The first response is always synchronous,
+			// and it would be nice to handle it immediately.
+			// However, there might be existing responses from the previous fetch,
+			// so we can't just immediately read from the channel and take the first result.
+			fme.fs.FetchRepeating(chanChanges)
+		case resp := <-chanChanges:
+			// Canceled errors are ignored,
+			// since they happen normally when stopping a request.
+			if isCanceledError(resp.err) {
+				break
+			}
+			if resp.err != nil {
+				fme.chanFetchRepeatingError <- resp.err
+				close(fme.chanFetchRepeatingError)
+				fme.chanFetchRepeatingError = nil
+				fme.fs.Close()
+				break
+			}
+			fme.handleFieldChanges(resp.changes)
+		}
+	}
+}
+
+// handleFieldChanges updates the list of fields the client currently knows about
+// according to changes made elsewhere (e.g. from fields changed from a script response or from a ListFields request).
+func (fm *fieldManagerExecutor) handleFieldChanges(changes *apppb2.FieldsChangeUpdate) {
+	for _, created := range changes.Created {
+		if created.TypedTicket.Type == "Table" {
+			fieldId := fieldId{appId: created.ApplicationId, fieldName: created.FieldName}
+			fm.tables[fieldId] = newTableHandle(fm.client, created.TypedTicket.Ticket, nil, 0, false)
+		}
+	}
+
+	for _, updated := range changes.Updated {
+		if updated.TypedTicket.Type == "Table" {
+			fieldId := fieldId{appId: updated.ApplicationId, fieldName: updated.FieldName}
+			fm.tables[fieldId] = newTableHandle(fm.client, updated.TypedTicket.Ticket, nil, 0, false)
+		}
+	}
+
+	for _, removed := range changes.Removed {
+		if removed.TypedTicket.Type == "Table" {
+			fieldId := fieldId{appId: removed.ApplicationId, fieldName: removed.FieldName}
+			delete(fm.tables, fieldId)
+		}
+	}
+}
+
+// listOpenableTables returns a list of the (global) tables that can be opened with OpenTable.
+func (fm *fieldManagerExecutor) listOpenableTables() []string {
+	var result []string
+
+	for id := range fm.tables {
+		if id.appId == "scope" {
+			result = append(result, id.fieldName)
+		}
+	}
+
+	return result
+}
+
+// fieldManager is an interface to fieldManagerExecutor.
```

Comment:
What does this mean?  It isn't an interface.  It is concrete.  Imagine someone new came to this code and needed to modify it.  What should the doc for this struct be?